R/utils_assert.R

```diff
@@ -3,4 +3,25 @@ assert_is <- function(x, what, name = deparse(substitute(x))) {
     stop(sprintf("'%s' must be a %s", name,
                  paste(what, collapse = " / ")), call. = FALSE)
   }
+  invisible(x)
+}
+
+
+assert_integer <- function(x, name = deparse(substitute(x)),
+                           what = "integer") {
+  if (!(is.integer(x))) {
+    usable_as_integer <- is.numeric(x) && (max(abs(round(x) - x)) < 1e-8)
```

Comment:
Is this a machine epsilon, and why 1e-8?
Looks fine though - does it need to be rounded above too?

Reply:
just pulled in from another package - sqrt(.Machine$double.eps) would be nicer at the cost of computing that every time. Rounding on the way out is a good shout, will implement that